fms/models/gpt_bigcode.py

```diff
@@ -331,10 +332,14 @@ def forward(
 
         preds = self.head(output)
 
+        out = [preds]
         if use_cache:
-            return preds, cache
-        else:
-            return preds
+            out.append(cache)
+        if include_embeds:
+            out.append(output)
+        if len(out) == 1:
+            return out[0]
+        return out
```

Comment:
I'd add documentation to the forward method explaining the order in which outputs are returned in case we return more than 1, same for all other models